package_test.go

```diff
@@ -274,64 +274,77 @@ func (s *PackageSuite) TestIterGetErrors(c *C) {
 		query   string
 		types   []any
 		inputs  []any
-		outputs [][]any
+		outputs []any
 		err     string
 	}{{
 		summary: "nil parameter",
 		query:   "SELECT * AS &Person.* FROM person",
 		types:   []any{Person{}},
 		inputs:  []any{},
-		outputs: [][]any{{nil}},
+		outputs: []any{nil},
 		err:     "cannot get result: need map or pointer to struct, got nil",
 	}, {
 		summary: "nil pointer parameter",
 		query:   "SELECT * AS &Person.* FROM person",
 		types:   []any{Person{}},
 		inputs:  []any{},
-		outputs: [][]any{{(*Person)(nil)}},
+		outputs: []any{(*Person)(nil)},
 		err:     "cannot get result: got nil pointer",
 	}, {
 		summary: "non pointer parameter",
 		query:   "SELECT * AS &Person.* FROM person",
 		types:   []any{Person{}},
 		inputs:  []any{},
-		outputs: [][]any{{Person{}}},
+		outputs: []any{Person{}},
 		err:     "cannot get result: need map or pointer to struct, got struct",
 	}, {
 		summary: "wrong struct",
 		query:   "SELECT * AS &Person.* FROM person",
 		types:   []any{Person{}},
 		inputs:  []any{},
-		outputs: [][]any{{&Address{}}},
+		outputs: []any{&Address{}},
 		err:     `cannot get result: type "Address" does not appear in query, have: Person`,
 	}, {
 		summary: "not a struct",
 		query:   "SELECT * AS &Person.* FROM person",
 		types:   []any{Person{}},
 		inputs:  []any{},
-		outputs: [][]any{{&[]any{}}},
+		outputs: []any{&[]any{}},
 		err:     "cannot get result: need map or pointer to struct, got pointer to slice",
 	}, {
 		summary: "missing get value",
 		query:   "SELECT * AS &Person.* FROM person",
 		types:   []any{Person{}},
 		inputs:  []any{},
-		outputs: [][]any{{}},
+		outputs: []any{},
 		err:     `cannot get result: type "Person" found in query but not passed to get`,
 	}, {
 		summary: "multiple of the same type",
 		query:   "SELECT * AS &Person.* FROM person",
 		types:   []any{Person{}},
 		inputs:  []any{},
-		outputs: [][]any{{&Person{}, &Person{}}},
+		outputs: []any{&Person{}, &Person{}},
 		err:     `cannot get result: type "Person" provided more than once, rename one of them`,
 	}, {
 		summary: "multiple of the same type",
 		query:   "SELECT name AS &M.* FROM person",
 		types:   []any{sqlair.M{}},
 		inputs:  []any{},
-		outputs: [][]any{{&sqlair.M{}, sqlair.M{}}},
+		outputs: []any{&sqlair.M{}, sqlair.M{}},
 		err:     `cannot get result: type "M" provided more than once, rename one of them`,
+	}, {
+		summary: "output expr in a with clause",
+		query: `WITH averageID(avgid) AS
+		        (SELECT &Person.id
+		        FROM person)
+		        SELECT id
+		        FROM person, averageID
+		        WHERE id > averageID.avgid
+				LIMIT 1`,
+		types:   []any{Person{}},
+		inputs:  []any{},
+		outputs: []any{&Person{}},
+		err:     `cannot get result: column "id" for struct "Person" not found in results`,
```

Comment:
As we discussed, the error here is unclear since the real issue is the &Person.id is being used in a wrong context. Maybe something along the lines of:
```
cannot get result: query uses &Person.id outside of result context
```
Also, FWIW the query right above seems broken a bit too much.

Reply:
Changed.